File: OCP/correct/PaymentConfig.cs
using System;
using System.Collections.Generic;

public class PaymentConfig
{
    private readonly Dictionary<string, Action<double>> strategies;

    public PaymentConfig()
    {
        strategies = new Dictionary<string, Action<double>>
        {
            { "CREDIT_CARD", amount => Console.WriteLine("Paying with card") },
            { "UPI", amount => Console.WriteLine("Paying with UPI") }
        };
    }

    public void Pay(string method, double amount)
    {
        if (!strategies.TryGetValue(method, out var strategy))
        {
            throw new ArgumentException($"Unsupported payment method: {method}");
        }

        strategy(amount);
    }
}

File: OCP/correct/PaymentConfig.ts
type PaymentStrategy = (amount: number) => void;

class PaymentConfig {
    private readonly strategies: Map<string, PaymentStrategy>;

    constructor() {
        this.strategies = new Map<string, PaymentStrategy>([
            ["CREDIT_CARD", (amount) => console.log("Paying with card")],
            ["UPI", (amount) => console.log("Paying with UPI")]
        ]);
    }

    pay(method: string, amount: number): void {
        const strategy = this.strategies.get(method);

        if (!strategy) {
            throw new Error(`Unsupported payment method: ${method}`);
        }

        strategy(amount);
    }
}
